src/smw/player_components/PlayerInvincibility.cpp

```diff
@@ -42,21 +40,19 @@ bool PlayerInvincibility::is_on() const
 void PlayerInvincibility::update(CPlayer& player)
 {
     if (invincible) {
-        player.animationtimer++;
-
-        if ((player.animationtimer > 3 && timer < 480) || player.animationtimer > 6) {
-            player.animationtimer = 0;
-
-            player.animationstate += 32;
-            if (player.animationstate > 96)
-                player.animationstate = 0;
-        }
-
         if (++timer > 580) {
-            player.animationstate = 0;
-            player.animationtimer = 0;
             timer = 0;
             invincible = false;
         }
     }
 }
+
+PlayerPalette PlayerInvincibility::getPlayerPalette() const
+{
+    PlayerPalette invincibility_animation_states[4] = {normal, invincibility_1, invincibility_2, invincibility_3};
+    if (timer < 480) {
+        return invincibility_animation_states[(timer / 4) % 4];
+    } else {
+        return invincibility_animation_states[((timer - 4) / 7) % 4];
+    }
```

Comment:
Is there any way we could make the indices more readable, that would explain _why_ these numbers are used? for example,

- If you use `std::array` for the animation states, you can then use its `size()` in the `% 4` part
- The `/ 4` and `/ 7` parts seem to control the speed of the animation – you could move them to a named variable
- Why do we subtract `- 4` in the second branch?

Reply:
Addressed.